scalafmt-core/native/src/main/scala/org/scalafmt/internal/RegexCompat.scala

```diff
@@ -0,0 +1,233 @@
+package org.scalafmt.internal
+
+import java.util.regex.Pattern
+
+import scala.util.matching.Regex
+
+/* Before text matching (?=re), after text matching (?<=re)
+ * and more are incompatible in Scala Native, so custom functions
+ * have to be used.
+ * Scala Native uses an implementation of:
+ * https://github.com/google/re2/wiki/Syntax
+ */
+
+object RegexCompat {
+
+  /* Replaces '\\h', which is incompatible in Scala Native.
+   * Does not check correctness of the input regex string.
+   */
+  private def fixHorizontalSpaceInRegex(reg: String) = {
+
+    @inline
+    val replacingInsideClass =
+      "\t \u00A0\u1680\u180E\u2000-\u200A\u202F\u205F\u3000"
+
+    @inline
+    val replacingOutsideClass = s"[$replacingInsideClass]"
+
+    val sb = new StringBuilder()
+    var isInClass = false
+    var isEscaped = false
+
+    for (char <- reg) char match {
+      case '\\' if !isEscaped => isEscaped = true
+      case 'h' if isEscaped =>
+        sb.append(if (isInClass) replacingInsideClass else replacingOutsideClass)
+        isEscaped = false
+      case '[' if !isEscaped =>
+        sb.append('[')
+        isInClass = true
+      case ']' if !isEscaped =>
+        sb.append(']')
+        isInClass = false
+      case other =>
+        if (isEscaped) {
+          isEscaped = false
+          sb.append('\\')
+        }
+        sb.append(other)
+    }
+    sb.toString()
+  }
```

Comment:
This is mostly a reply to a related comment from a previous PR. This method was written mostly because I found `\\h` replacements specifically very annoying, unintuitive and ugly. But the regexes here have many more differences unfortunately, so just supplying this method to JVM style regexes would not work unfortunately.